The Marauder's Map - Nadine Adnane, Nick Liu, & Insha Lakhani

nadnane opened the pull request:
### [Marauder's Map Design Doc](https://github.com/liunicholas6/marauders-map/blob/main/proposal/proposal.md)

![5e059eab2773341bb23c23ab5e6acc90](https://github.com/CIS-566-Fall-2023/final-project/assets/29757481/702f7666-5a32-4901-9f18-b1c949d8b8e9)

Commit:
Initial Commit

nadnane committed:
Design Doc WIP

nadnane committed:
Added free-body diagram

Commit:
Proposal done

nadnane committed:
Formatting fix + add logo

liunicholas6 committed:
We are definitely using unity; merging into main

nadnane committed:
Add fonts, logo, and update map generator

nadnane committed:
UI WIP

Added canvas, adjusted camera and map generator settings

liunicholas6 committed:
Merging room creation

liunicholas6 committed:
Merge

nadnane committed:
WIP scene setup, interactivity, post-processing

Added post-processing package, fixed map animation and UVs, added table, wand, and canvas for user prompt.

liunicholas6 committed:
Curve to point stream

liunicholas6 committed:
Building Generator top-level class created; depedencies updated

liunicholas6 committed:
Walls

liunicholas6 committed:
Arc Curve

liunicholas6 committed:
Moved rectangle; created sector and sector region

liunicholas6 committed:
Created tower runner and debug sector

liunicholas6 committed:
Patched ToPointStream

liunicholas6 committed:
Fixed DebugSector

liunicholas6 committed:
Patched sector to be struct and TowerRunner left/right to be correct

inshalak committed:
working sprite

liunicholas6 committed:
Progress

liunicholas6 committed:
Progress

liunicholas6 committed:
Progress

inshalak committed:
added bezier

liunicholas6 committed:
Merge branch 'building-generation' of github.com:liunicholas6/marauders-map into building-generation

liunicholas6 committed:
Debug binary partition

liunicholas6 committed:
Commented Tower Runner

liunicholas6 committed:
Restore TowerRunner

liunicholas6 committed:
Tower complete

liunicholas6 committed:
Tower Complete

liunicholas6 committed:
Merge branch 'building-generation'

nadnane committed:
WIP render text, remove necessary files

liunicholas6 committed:
Room walls have thickness

nadnane committed:
Add new map scene (old one is corrupted)

nadnane committed:
Mark anims as legacy

nadnane committed:
Import TMP essentials

nadnane committed:
Merge branch 'main' into rendering

inshalak committed:
updates

inshalak committed:
updates + added files

liunicholas6 committed:
Merge branch 'sprites'

nadnane committed:
Added URP and updated shaders

liunicholas6 committed:
Post-merge fixes

liunicholas6 committed:
LineCurve -> ICurve patch

liunicholas6 committed:
Patch

nadnane committed:
Add bookshelves and candle glow

nadnane committed:
Merge branch 'main' into rendering

nadnane committed:
Add post processing

inshalak committed:
HE ZOOM IN DA ROOM

liunicholas6 committed:
Added tangent angle

liunicholas6 committed:
Tangent angle patch

nadnane committed:
Draw map after unfold animation ends

liunicholas6 committed:
Reverting wanderers

nadnane committed:
Remove extra map scene

nadnane committed:
Fix missing prefab issue

nadnane committed:
Attach camera to UI Canvas

nadnane committed:
Create MapText.png

liunicholas6 committed:
Line textures; line render works for arcs

nadnane committed:
Footprint trail WIP

liunicholas6 committed:
Lines and prefabs

liunicholas6 committed:
Text mapping good. Block + Tower

liunicholas6 committed:
Merge branch 'rendering'

inshalak commented:
[Presentation](https://www.canva.com/design/DAF2M6iz1rE/t4Pxd4sO3kSRoSdICaY6Fw/view?utm_content=DAF2M6iz1rE&utm_campaign=designshare&utm_medium=link&utm_source=editor)

nadnane committed:
Footprints working!

Added footprint particle system! Only thing left is to make the footsteps rotate based off of the navigation direction.

nadnane committed:
Replace default README with proposal

nadnane committed:
Update README.md

inshalak committed:
footsteps fixed

inshalak committed:
Merge branch 'main' of https://github.com/liunicholas6/marauders-map

inshalak committed:
footsteps better

inshalak committed:
adding in final texture

inshalak committed:
adding texture meta

liunicholas6 committed:
Cleanup formatting

liunicholas6 committed:
Fixed Bezier tangent typo

liunicholas6 committed:
Footstep orientation in hallways

liunicholas6 committed:
Random walk without lag implemented (currently teleports across rooms)

liunicholas6 committed:
Footstep orientation

liunicholas6 committed:
Walk through rooms; switch to queue + purely random walk to lay groundwork for DFS

liunicholas6 committed:
DFS for paths between random points

liunicholas6 committed:
DFS now adds edges for rooms

liunicholas6 committed:
ArcCurve tangent patch, added wanderers back

nadnane committed:
Added 4 line material slots

nadnane committed:
Fix interactive text prompt

nadnane committed:
Add presentation links

nadnane commented:
Presentation links

PPT online:
https://penno365-my.sharepoint.com/:p:/g/personal/nadnane_upenn_edu/EZ8V0aZEFTpJiXzar6lDyIwBO8fO57vTVqHnb1Q3ooMjwA?e=QFgmQL


Presentation PDF:
https://drive.google.com/file/d/1ZAWCUCIz_tjufm5roYYtgfrHDat_GOop/view?usp=drive_link

Video demo:
https://drive.google.com/file/d/1_NIrF6KQkl-19mpZpq_-dLrtlod3lRqT/view?usp=drive_link

nadnane commented:
https://drive.google.com/file/d/151bN_SmVmYf8pYRNdMN71J27yp8UWNIx/view?usp=drive_link, https://drive.google.com/file/d/1MvLcDiij6jcQsvgciwN9479qnozQwp3-/view?usp=drive_link, https://drive.google.com/file/d/1ZAWCUCIz_tjufm5roYYtgfrHDat_GOop/view?usp=drive_link, https://drive.google.com/file/d/1_NIrF6KQkl-19mpZpq_-dLrtlod3lRqT/view?usp=drive_link
